lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1622,9 +1633,10 @@ public function removeFromIdentityMap($entity)
 
         $className = $classMetadata->rootEntityName;
 
+        unset($this->oidMap[$oid]);
+        unset($this->readOnlyObjects[$oid]);
```

Comment:
There's exactly the same issue with the read only objects: they can points at dead entities. We have to clear the oid even if the entity is not found into the identityMap.